notus/scanner/loader/gpg_sha_verifier.py

```diff
@@ -0,0 +1,71 @@
+import hashlib
+import os
+from pathlib import Path
+from typing import Callable, Dict, Optional
+
+from gnupg import GPG
+
+
+def __default_gpg_home() -> GPG:
+    """
+    __defaultGpgHome tries to load the variable 'GPG_HOME' or to guess it
+    """
+    manual = os.getenv("GPG_HOME")
+
+    home = Path(manual) if manual else Path(os.getenv("HOME", "")) / ".gnupg"
+    return GPG(gnupghome=f"{home.absolute()}")
+
+
+def gpg_sha256sums(
+    hash_file: Path, gpg: Optional[GPG] = None
+) -> Dict[str, str]:
+    """
+    gpg_sha256sums verifies given hash_file with a asc file
+
+    This functions assumes that the asc file is in the same directory as the
+    hashfile and has the same name but with the suffix '.asc'
+    """
+
+    # when doing that via paramater list it is loading eagerly on import
+    # which may fail on some systems
+    if not gpg:
+        gpg = __default_gpg_home()
+    if not hash_file.is_file():
+        raise Exception(f"{hash_file.absolute()} is not a file")
+    asc_path = hash_file.parent / f"{hash_file.name}.asc"
+    with asc_path.open(mode="rb") as f:
+        verified = gpg.verify_file(f, f"{hash_file.absolute()}")
+        if not verified:
+            raise Exception(f"verification of {hash_file.absolute()} failed")
```

Comment:
Could you please use an error from https://github.com/greenbone/notus-scanner/blob/main/notus/scanner/errors.py or alternatively create a new error class?

Reply:
Ok, I would rather create a new one since this code will initially duplicated in ospd-openvas until we have a module project ready :).